okhttp/src/main/kotlin/okhttp3/internal/connection/RealCall.kt

```diff
@@ -233,17 +245,21 @@ class RealCall(
 
   /** Finds a new or pooled connection to carry a forthcoming request and response. */
   internal fun initExchange(chain: RealInterceptorChain): Exchange {
-    check(expectMoreExchanges) { "released" }
-    check(!responseBodyOpen)
-    check(!requestBodyOpen)
+    synchronized(this) {
+      check(expectMoreExchanges) { "released" }
+      check(!responseBodyOpen)
+      check(!requestBodyOpen)
+    }
```

Comment:
Yep, failures were my fault. I move the assignments on line 259 up here to save a synchronized block, and that was broken if the lines in-between throw! Argh!

I believe the update doesn’t have behavior changes, other than introduced locking.